sklearn/ensemble/gradient_boosting.py

```diff
@@ -1493,9 +1493,9 @@ def fit(self, X, y, sample_weight=None, monitor=None):
                         raise ValueError(msg)
                     except ValueError as e:
                         if 'not enough values to unpack' in str(e):  # pipeline
-                            raise ValueError(msg)
+                            raise ValueError(msg) from e
                         else:  # regular estimator whose input checking failed
-                            raise e
+                            raise
```

Comment:
Nit: Do not need the `else` here

Reply:
I personally prefer the whole if/else logic. It's clearer and has a more functional flavor.